node/src/chain/acceptor.rs

```diff
@@ -363,13 +471,18 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> Acceptor<N, DB, VM> {
                 info!("Slashed {}", slashed.to_base58())
             }
 
-            match Self::needs_update(blk.inner(), &txs) {
-                true => {
-                    let state_hash = blk.inner().header().state_hash;
-                    let new_prov = vm.get_provisioners(state_hash)?;
-                    provisioners_list.update_and_swap(new_prov)
-                }
-                false => provisioners_list.remove_previous(),
+            let selective_update = Self::selective_update(
+                blk.inner(),
+                &txs,
+                &vm,
+                &mut provisioners_list,
+            );
+
+            if let Err(e) = selective_update {
+                warn!("Resync provisioners due to {e:?}");
```

Comment:
This sounds like a unrecoverable error, not just a warning. Any additional details we can trace so we make this easier for debugging?

Reply:
Going to change to error!
All details are in the error stack